src/api/models/iuf/iuf-manifest-schema.yaml

```diff
@@ -373,6 +373,32 @@ properties:
                     this will default to the name of the tar file without the
                     ".tar.gz" extension.
                   type: string
+                recipe_type:
+                  description: >
+                    The recipe type being built.
+                  type: string
+                  examples:
+                  - "kiwi-ng"
+                  - "packer"
+                linux_distribution:
+                  description: >
+                    The Linux distribution for the image being built.
+                  type: string
+                  examples:
+                  - "sles12"
+                  - "sles15"
+                  - "centos7"
+                template_dictionary:
+                  description: >
+                    Key-value pairs to be templated into the recipe when
+                    building the image.
+                  type: object
+                  properties: {}
```

Comment:
I do not believe this is necessary. Testing:

```
C02ZH00ALVCJ: cray-nls (CASM-3594-update-schema *%) $ git diff
diff --git a/src/api/models/iuf/iuf-manifest-schema.yaml b/src/api/models/iuf/iuf-manifest-schema.yaml
index 264fd61..1b01b03 100644
--- a/src/api/models/iuf/iuf-manifest-schema.yaml
+++ b/src/api/models/iuf/iuf-manifest-schema.yaml
@@ -393,7 +393,6 @@ properties:
                     Key-value pairs to be templated into the recipe when
                     building the image.
                   type: object
-                  properties: {}
                   additionalProperties:
                     type: string
                 md5sum:
C02ZH00ALVCJ: cray-nls (CASM-3594-update-schema *%) $ go build
C02ZH00ALVCJ: cray-nls (CASM-3594-update-schema *%) $ cat examples/iuf-product-manifest/ims-artifacts.yaml
---
name: myproduct
version: 1.2.3
iuf_version: ^0.3.0

content:
  ims:
    recipes:
    - path: myrecipe
      recipe_type: kiwi-ng
      linux_distribution: sles15
      template_dictionary:
        fruit: apple
        vegetable: kale
      md5sum: d3b07384d113edec49eaa6238ad5ff00
C02ZH00ALVCJ: cray-nls (CASM-3594-update-schema *%) $ ./cray-nls validate examples/iuf-product-manifest/ims-artifacts.yaml
```

Is it also worth checking in an example file like the above? It's kind of helpful for validating schema as you're writing it, and maybe we could incorporate these into some tests?

Reply:
Ah, good to see that this still works. I interpreted the `additionalProperties` description in the [core JSON Schema spec](https://json-schema.org/draft/2020-12/json-schema-core.html#additionalProperties) to mean that the `properties` attribute was necessary, but since there are no required attributes, I wanted it to explicitly be empty.
> The behavior of this keyword depends on the presence and annotation results of "properties" and "patternProperties" within the same schema object. Validation with "additionalProperties" applies only to the child values of instance names that do not appear in the annotation results of either "properties" or "patternProperties".